lighthouse-core/gather/driver.js

```diff
@@ -630,6 +630,60 @@ class Driver {
     return new LHElement(targetNode, this);
   }
 
+  /**
+   * @param {string} handle
+   * @param {{timeout: number}=} options,
+   * @return {Promise<string>}
+   */
+  async readIOStream(handle, options = {timeout: 5000}) {
+    const startTime = Date.now();
+
+    let ioResponse;
+    let data = '';
+    while (!ioResponse || !ioResponse.eof) {
+      if (Date.now() - startTime > options.timeout) {
+        throw new Error('Waiting for the end of the IO stream exceeded the allotted time.');
+      }
+      ioResponse = await this.sendCommand('IO.read', {handle});
+      data = data.concat(ioResponse.data);
+    }
+
+    if (ioResponse.base64Encoded) {
+      data = Buffer.from(data, 'base64').toString('utf-8');
+    }
+    return data;
+  }
+
+  /**
+   * @param {string} url
+   * @return {Promise<string>}
+   */
+  async fetchResourceOverProtocol(url) {
+    const milestone = await this.getBrowserVersion().then(v => v.milestone);
```

Comment:
Can you move this to `Fetcher`? with FR we're starting to chip away at the God class that is driver :)

Reply:
Leave `readIOStream` here? It doesn't need to be specific to a resource fetcher.

Reply:
sure